plugin/storage/grpc/factory_test.go

```diff
@@ -120,7 +121,11 @@ func TestNewFactoryError(t *testing.T) {
 
 	t.Run("client", func(t *testing.T) {
 		// this is a silly test to verify handling of error from grpc.NewClient, which cannot be induced via params.
-		f, err := NewFactoryWithConfig(Config{}, metrics.NullFactory, zap.NewNop(), componenttest.NewNopHost())
+		f, err := NewFactoryWithConfig(Config{
+			ClientConfig: configgrpc.ClientConfig{
+				Endpoint: "http://test.com",
```

Comment:
this doesn't look like a quality fix to me. Giving it a fake address may cause undesired connection, which could explain go leaks.

Reply:
@yurishkuro it doesn't matter what this is for the unit tests it just shouldn't be blank to avoid the empty build error - changed it to just a random string now

Reply:
@yurishkuro sorry! you were right - i changed the ports to be `:0` and it looks good now